### Hypothesis Testing

Hypothesis is a methodology which is used to prove or dissapprove the stataement with the help of Stastical Analysis.

In this assignment we have to Frame 10 Hypothesis in this dataset.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
dataset = pd.read_csv("general_data.csv")
dataset = dataset.dropna()

In [3]:
dataset

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404  Female  ...                 1.0      Y                18              8   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [4]:
dataset['Age'].mean()

36.933363760839796

1st Hypothesis 

Here we take the mean age of all the employees be 37


so h0:u <= 37


and ha:u > 3

In [5]:
age_dataset = dataset['Age']

In [6]:
samm_age = age_dataset[:50]

In [7]:
samm_age.mean()

36.32

After checking the sample of 50 employees it is found that the mean is 36.32 so opur Null Value is
Accepted

In [8]:
dataset['StandardHours'].mode()

0    8
dtype: int64

2nd Hypothesis 

Here we take the mode pf  Standard Hours  of all the employees be 8 hrs
Now lets check the Hypothesis  of this by taking a sample of 70 Employees.

so h0:u <= 0


and ha:u > 0

In [9]:
Hrs = dataset['StandardHours']

In [10]:
std_hours = Hrs[:69]

In [11]:
std_hours.mode()

0    8
dtype: int64

After checking the sample of 70 Employees it is found that the mode is 0 so opur Null Value is
Accepted

In [12]:
dataset['YearsAtCompany'].mean()

7.010497489730716

3rd Hypothesis

Here we take the mean YearsatCompany of all the Employees  be 7 years
Now lets check the Hypothesis  of this by taking a sample of 70 persons.

so h0:u <= 7


and ha:u > 7

In [13]:
years_at_company = dataset['YearsAtCompany']

In [14]:
years_sam = years_at_company[:69]

In [15]:
years_sam.mean()

8.072463768115941

After checking the sample of 70 employees it is found that the mean is 8 so our Null Value is rejected this time.

In [16]:
dataset["YearsSinceLastPromotion"].mean()

2.191693290734824

4th Hypothesis 

Here we take the mean of Years with Year since Last Promotion of all the people  be 2 years
Now lets check the Hypothesis  of this by taking a sample of 70 employees.

so h0:u <= 2


and ha:u > 2

In [17]:
year_promotion = dataset["YearsSinceLastPromotion"]

In [18]:
sam_year_promoion = year_promotion[:69]

In [19]:
sam_year_promoion.mean()

2.8405797101449277

After checking the sample of 70 employees it is found that the mean is 2.84 so our Null Value is rejected this time.

In [20]:
dataset['StockOptionLevel'].skew()

0.9672626781773564

5th Hypothesis 

Here we take the Skewness of Stocks Option Level of all the people  be 0.96 years
Now lets check the Hypothesis  of this by taking a sample of 70 persons.

so h0:u <= 0.96


and ha:u > 0.96

In [21]:
Stock_option_Level = dataset['StockOptionLevel']

In [22]:
sam_Stock_option_Level = Stock_option_Level[:69]

In [23]:
sam_Stock_option_Level.skew()

0.8594283325245387

After checking the sample of 70 employees it is found that the skew is 0.85 so our Null Value is accepted

In [24]:
dataset['PercentSalaryHike'].median()

14.0

6th Hypothesis

Here we take the median of Stocks of Percent Salary Hike of all the Employees be 14.0 percent
Now lets check the Hypothesis  of this by taking a sample of 70 persons.

so h0:u <= 14.0


and ha:u > 14.0

In [25]:
percent_sal_hike = dataset['PercentSalaryHike']

In [26]:
sam_percent_sal_hike = percent_sal_hike[:69]

In [27]:
percent_sal_hike.median()

14.0

After checking the sample of 70 employees it is found that the mean of percent salary hike is 14.0 so our Null Value is accepted

In [28]:
dataset['YearsWithCurrManager'].std()

3.569674079459996

7th Hypothesis 

Here we take the standard deviation of Stocks of Years with Current of all the employees be 3.56  percent
Now lets check the Hypothesis  of this by taking a sample of 70 employees.

so h0:u <= 3.56


and ha:u > 3.56

In [29]:
year_with_cur_man = dataset['YearsWithCurrManager']

In [30]:
sam_year_with_cur_man = year_with_cur_man[:69]

In [31]:
sam_year_with_cur_man.std()

3.444542986426028

After checking the sample of 70 employees it is found that the standard deviation of Years with Current Manager is 3.44 so our Null Value is accepted

In [32]:
dataset['YearsAtCompany'].var()

37.56893794007587

8th Hypothesis 

Here we take the Variance of Stocks of Years at company of all the employees be 37.56  percent
Now lets check the Hypothesis  of this by taking a sample of 70 persons.

so h0:u <= 3.56


and ha:u > 3.56

In [33]:
Years_at_company = dataset['YearsAtCompany']

In [34]:
sam_Years_at_company = Years_at_company[:69]

In [35]:
sam_Years_at_company.var()

44.77408354646207

After Checking its variance it is out in the boundary of Rejection Area so we reject this value.

In [36]:
dataset['TotalWorkingYears'].kurt()

0.9093160190773464

9th Hypothesis

Here We will Chaeck Kurtosis of Total Working Years which is 0.90
for Hypothesis Check we will check it with 70 data sample.

H0:u <= 0.90

Ha:u > 0.90

In [37]:
total_work_years = dataset['TotalWorkingYears']

In [38]:
sam_total_work_years = total_work_years[:69]

In [39]:
sam_total_work_years.kurt()

0.9220200363993944

After checking the Kurtosis from the sample of 70 Employees it lies outside the boundary so cannot accept this value, we are rejecting this

In [40]:
dataset['DistanceFromHome'].mean()

9.198995892286627

10th Hypothesis

Here We will Check the mean of Distance From Home of Employees is 9.19
for Hypothesis Check we will check it with 70 data sample.

H0:u <= 9.19

Ha:u > 9.19

In [41]:
distance_home = dataset['DistanceFromHome']

In [42]:
sam_distance_home = distance_home[:69]

In [43]:
sam_distance_home.mean()

7.942028985507246

So, after Hypothesis our value comes 7.94 so it comes under 9.19 so our null test value is accepted